optimum/modeling_base.py

```diff
@@ -415,13 +415,6 @@ def from_pretrained(
                 trust_remote_code=trust_remote_code,
             )
 
-        if not export and trust_remote_code:
-            logger.warning(
-                "The argument `trust_remote_code` is to be used along with export=True. It will be ignored."
```

Comment:
`trust_remote_code` will be used to load the configuration even when `export=False`